cli/azd/resources/apphost/templates/resources.bicept

```diff
@@ -125,44 +129,56 @@ resource {{bicepName $name}} 'Microsoft.App/containerApps@2023-05-02-preview' =
     environmentId: containerAppEnvironment.id
     configuration: {
       activeRevisionsMode: 'Single'
-{{if $value.Dapr}}
+{{- if $value.Dapr}}
       dapr: {
         appId: {{$value.Dapr.AppId}}
-{{if $value.Dapr.AppPort}}
+{{- if $value.Dapr.AppPort}}
         appPort: {{$value.Dapr.AppPort}}
-{{end}}
-{{if $value.Dapr.AppProtocol}}
+{{- end}}
+{{- if $value.Dapr.AppProtocol}}
         appProtocol: {{$value.Dapr.AppProtocol}}
-{{end}}
-{{if $value.Dapr.EnableApiLogging}}
+{{- end}}
+{{- if $value.Dapr.EnableApiLogging}}
         enableApiLogging: {{$value.Dapr.EnableApiLogging}}
-{{end}}
+{{- end}}
         enabled: true
-{{if $value.Dapr.HttpMaxRequestSize}}
+{{- if $value.Dapr.HttpMaxRequestSize}}
         httpMaxRequestSize: {{$value.Dapr.HttpMaxRequestSize}}
-{{end}}
-{{if $value.Dapr.HttpReadBufferSize}}
+{{- end}}
+{{- if $value.Dapr.HttpReadBufferSize}}
         httpReadBufferSize: {{$value.Dapr.HttpReadBufferSize}}
-{{end}}
-{{if $value.Dapr.LogLevel}}
+{{- end}}
+{{- if $value.Dapr.LogLevel}}
         logLevel: {{$value.Dapr.LogLevel}}
-{{end}}
+{{- end}}
       }
-{{end}}
-{{if $value.Ingress}}      
+{{- end}}
+{{- if $value.Ingress}}
       ingress: {
         external: {{$value.Ingress.External}}
         targetPort: {{$value.Ingress.TargetPort}}
         transport: '{{$value.Ingress.Transport}}'
+{{- if ne $value.Ingress.Transport "tcp" }}
```

Comment:
This may be a bug fix for something else, but just confirming the intent of the change.

Reply:
Indeed - this was fixing a pre-existing issue that I discovered while doing some end to end testing myself.  The problem was before we would always set `allowInsecure` (which tells ACA if the Envoy proxy that fronts each host should allow HTTP on port 80 (`true`) or if it should instead preform an HTTP-HTTPS redirection (`false`).

For containers that expose just raw TCP ports, there is no load balancer and ACA fails the request if the `allowInsecure` value is set.  So this change ensures we exclude it on these types of ingress configurations.